### PR TITLE
fix: new api added for getting ciPipelineDetails by only ciPipelineId for CLI

### LINKED ISSUE #3578: Feature: API to update pre/post ci/cd using app_name and env_name
### 🔖 Feature description

API to update pre/post ci/cd using app id/name and env id/name.
Option of patch should also be given.

### 🎤 Pitch

Right now we have to get workflow id, ci pipeline id manually to update any pre/post ci using API but that doesn't work if we want to perform the operations in loop

### ✌️ Solution

Having an api that does it automatically using app id and env id can be helpful

### 🔄️ Alternative

_No response_

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/devtron-labs/devtron/blob/main/CODE_OF_CONDUCT.md)

AB#4048

### PR DESCRIPTION
**Pull request by prakash100198**
# Description

New api added for fetching ciPipelineDetails by just ciPipelineId
Fixes #3578 

## How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

## Checklist:

* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR requires documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] I have performed a self-review of my own code.
* [ ] I have commented my code, particularly in hard-to-understand areas.
* [ ] I have tested it for all user roles.
* [ ] I have added all the required unit/api test cases.

## Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: api/restHandler/app/BuildPipelineRestHandler.go [hunk 1]
```diff
@@ -1053,6 +1055,66 @@ func (handler PipelineConfigRestHandlerImpl) GetCIPipelineById(w http.ResponseWr
 	common.WriteJsonResp(w, err, ciPipeline, http.StatusOK)
 }
 
+func (handler PipelineConfigRestHandlerImpl) GetCIPipelineByPipelineId(w http.ResponseWriter, r *http.Request) {
+	token := r.Header.Get("token")
+	var ciPipelineId int
+	var err error
+	v := r.URL.Query()
+	pipelineId := v.Get("pipelineId")
+	if len(pipelineId) != 0 {
```
Comment (Kripansh) [inline api/restHandler/app/BuildPipelineRestHandler.go:1064]: what about zero length pipelineId ??

File: api/restHandler/app/BuildPipelineRestHandler.go [hunk 2]
```diff
@@ -1053,6 +1055,66 @@ func (handler PipelineConfigRestHandlerImpl) GetCIPipelineById(w http.ResponseWr
...
+	app, err := handler.pipelineBuilder.GetApp(ciPipeline.AppId)
+	if err != nil {
+		handler.Logger.Infow("service error, GetCIPipelineByPipelineId", "err", err, "appId", ciPipeline.AppId, "pipelineId", pipelineId)
+		common.WriteJsonResp(w, err, nil, http.StatusBadRequest)
+		return
+	}
+	ciPipeline.AppName = app.AppName
+	ciPipeline.AppType = app.AppType
+
+	resourceName := handler.enforcerUtil.GetAppRBACName(app.AppName)
```
Comment (Kripansh) [inline api/restHandler/app/BuildPipelineRestHandler.go:1091]: not optimized, instead use method `GetAppRBACNameByAppId`

File: api/restHandler/app/BuildPipelineRestHandler.go [hunk 3]
```diff
@@ -1053,6 +1055,66 @@ func (handler PipelineConfigRestHandlerImpl) GetCIPipelineById(w http.ResponseWr
...
+		common.WriteJsonResp(w, err, nil, http.StatusInternalServerError)
+		return
+	}
+	var environmentIds []int
+	for _, pipeline := range pipelineData {
+		environmentIds = append(environmentIds, pipeline.EnvironmentId)
+	}
+	if handler.appWorkflowService.CheckCdPipelineByCiPipelineId(ciPipelineId) {
+		for _, envId := range environmentIds {
+			envObject := handler.enforcerUtil.GetEnvRBACNameByCiPipelineIdAndEnvId(ciPipelineId, envId)
```
Comment (kripanshdevtron) [inline api/restHandler/app/BuildPipelineRestHandler.go:1108]: Why are we checking this ??

Comment (prakash100198) [reply api/restHandler/app/BuildPipelineRestHandler.go:1108]: Actually this ensures that the ci-pipeline that deploys to a particular env, for that particular user if he/she has update rights or not in that env, if not then that would be a forbidden, and also the same rbac was implemented in the method where we were getting ciPipelineDetails based on appId as well as pipelineId. So I took the same reference of RBAC implementation here also.

File: api/restHandler/app/BuildPipelineRestHandler.go (hunk 1 above)
Comment (prakash100198) [reply api/restHandler/app/BuildPipelineRestHandler.go:1064]: okay, this case was not handled, have handled it now

File: api/restHandler/app/BuildPipelineRestHandler.go (hunk 2 above)
Comment (prakash100198) [reply api/restHandler/app/BuildPipelineRestHandler.go:1091]: sure have used

**Review by vikramdevtron — APPROVED** [review]

**Pull request merged**